### PR TITLE
feat(tasks): Support creation of general/approval tasks

### PR DESCRIPTION
**Pull request by mattnguyen1**

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @mattnguyen1 has signed the CLA. Thanks for the pull request!

File: src/api/Feed.js [hunk 1]
```diff
@@ -679,6 +679,7 @@ class Feed extends Base {
         currentUser: User,
         message: string,
         assignees: SelectorItems,
+        taskType: TaskType,
```
Comment (ConradJChan) [inline src/api/Feed.js]: Update the doc

File: src/api/Feed.js [hunk 2]
```diff
@@ -749,13 +750,16 @@ class Feed extends Base {
...
-            task_type: 'GENERAL',
+            task_type: taskType,
             status: TASK_NEW_NOT_STARTED,
         };
 
         this.addPendingItem(this.id, currentUser, pendingTask);
 
-        const taskPayload = { name: message, due_at: dueAtString };
+        const taskPayload: TaskPayload = { name: message, due_at: dueAtString };
+        if (taskType) {
```
Comment (ConradJChan) [inline src/api/Feed.js]: According to the flow typing, it seems like `taskType` is required, is this if check needed?

File: src/elements/content-sidebar/AddTaskButton.js [hunk 3]
```diff
@@ -25,17 +26,29 @@ type Props = {| ...AddTaskButtonProps, ...PassThroughProps |};
...
         isTaskFormOpen: false,
+        taskType: null,
     };
 
     static defaultProps = {
         isDisabled: false,
     };
 
+    getMessageForModalTitle(taskType: ?TaskType): Object {
+        switch (taskType) {
```
Comment (ConradJChan) [inline src/elements/content-sidebar/AddTaskButton.js]: Should `TASK_TYPE_GENERAL` be the default case if `taskType` is not provided?

Comment (mattnguyen1) [reply src/elements/content-sidebar/AddTaskButton.js]: the backend creates an approval task if `taskType` is not provided

File: src/api/Feed.js (hunk 2 above)
Comment (mattnguyen1) [reply src/api/Feed.js]: will update doc to make it optional

Comment (alexkrolick) [reply src/api/Feed.js]: let's make it required - you should have to know what type of task you create

Comment (mattnguyen1) [reply src/api/Feed.js]: ah yeah that makes sense, I'll go ahead and make it required

File: src/elements/content-sidebar/AddTaskButton.js (hunk 3 above)
Comment (alexkrolick) [reply src/elements/content-sidebar/AddTaskButton.js]: we can the type non-optional/non-nullable here too

**Pull request merged**